### ISSUE TITLE
auth0 / comments in evals problem

### ISSUE THREAD

**Issue by jakobrosenberg** [PepsRyuu/nollup#121]
Using auth0 in a Nollup project produces the following error.

```
Uncaught SyntaxError: Invalid or unexpected token
    at Object.__impl (main.js:7247)
    at executeModuleImpl (main.js:185)
```

line 7247:
```javascript
eval('/*! *****************************************************************************\n\
Copyright (c) Microsoft Corporation. All rights reserved.\n\
Licensed under the Apache License, Version 2.0 (the "License"); you may not use\n\
...
***************************************************************************** */\n\
var e=function(t,n){return(...
```

**Comment by PepsRyuu** [comment]
Have a minimal reproducible example? Never used auth0. 🙂 I assume this can be corrected by escaping the forward slash?

**Comment by jakobrosenberg** [comment]
Sure, the auth branch here: https://github.com/roxiness/routify-starter/tree/auth

PS. This isn't high priority by any measure. Just figured it might have your interest. 🙂

**Comment by PepsRyuu** [comment]
It's because of this line here:

```
code = code
               .replace(/\\/g, '\\\\')
               .replace(/'/g, '\\\'')
               .replace(/(\r)?\n/g, '\\n\\\n')
               .replace(/\/\/# sourceMappingURL=(.*?)($|\n)/g, '') // remove existing sourcemapurls
```

```auth0``` has this in its code:

```
a=o.substring(i)+(r?"//# sourceMappingURL="+r:""),c=new
```
So the regex above cuts off the rest of the code. Need to revisit this.

**Comment by jakobrosenberg** [comment]
In case sourcemaps are always on a new line, you could add `^ *`.
```javascript
.replace(/^ *\/\/# sourceMappingURL=(.*?)($|\n)/g, '') // remove existing sourcemapurls
```

If sourcemaps always use `.map` extensions, you could add `\.map`.
```javascript
.replace(/^ *\/\/# sourceMappingURL=(.*\.map)($|\n)/g, '') // remove existing sourcemapurls
```

**Comment by PepsRyuu** [comment]
I'm thinking it might just be best to remove the replacement entirely. Seems like the last source mapping comment is the one that will trigger, but I'm not sure if this is reliable. Seems to hold true for both Chrome and Firefox.

**Comment by jakobrosenberg** [comment]
On one hand this sounds like something that could change from browser to browser or even version to version. On the other, it makes sense to actively target the very last sourcemap.

**Comment by PepsRyuu** [comment]
There's another problem with this as well, there's an overlap with the exports:

```
export default function (e) {...}export{fo as Auth0Client};
```
This doesn't work because of an overlap with ```magic-string```. It gets transformed into:

```
__e__('default', function (e) {};__e__('Auth0Client', fo);
```
This is because normally there's a bracket appended at the position where the second export begins, but because there's another export beginning at the same position, that overrides the appended bracket. This is going to be a tricky one to resolve.

**Comment by PepsRyuu** [comment]
This PR makes the app work for me: https://github.com/PepsRyuu/nollup/pull/125 

Would be great if you can confirm. :)

**Comment by jakobrosenberg** [comment]
Works for me too. Hope we're not testing the same repo. 😅

**Comment by rturnq** [comment]
Hate to resurrect an old issue but it seemed most appropriate as I just ran into this exact same issue importing auth0. It looks like the changes that previously fixed it (#125) were somewhat wiped out by #173.

I'm not sure if the previous fix can still work but I was able to get passed it by altering the regex to only match the final source map declaration using a negative lookbehind.

Here is an example: https://regex101.com/r/DfFZhf/1

**Comment by PepsRyuu** [comment]
Whoops, how did that get back in there? 🤦‍♂️ My bad.

If that line is removed does it still keep it fixed? I'd prefer not to do any messy regex stuff as it seems completely unreliable.

**Comment by rturnq** [comment]
Yep, that works. Probably a much safer approach.

**Comment by PepsRyuu** [comment]
Regression fixed in ```0.15.3```.
